src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -726,4 +750,93 @@ public static void constructIonStruct(IonStruct fields, IonList annotations, Ion
             writer.stepOut();
         }
     }
+
+    /**
+     * Write random Ion Lists which conform with the constraints in Ion Schema.
+     * @param size specifies the size in bytes of the generated file.
+     * @param format is the format of the generated file, select from set (ion_text | ion_binary).
+     * @param path the destination of the generated file.
+     * @param constraintStruct is an IonStruct which contains the top-level constraints in Ion Schema.
+     * @param annotations contains the information of annotations, both the constraint of the annotation and the value of the annotation.
+     * @throws Exception if errors occur when writing data.
+     */
+    public static void writeRandomListValues(int size, String format, String path, IonStruct constraintStruct, Map <String, Object> annotations) throws Exception {
+        File file = new File(path);
+        IonType type = IonType.LIST;
+        try (IonWriter writer = WriteRandomIonValues.formatWriter(format, file)) {
+            WriteRandomIonValues.writeRequestedSizeFile(size, writer, file, type, 0, null, null, null, null, null, annotations, constraintStruct);
+            WriteRandomIonValues.printInfo(path);
+        }
+    }
+
+    /**
+     * Construct Ion List based on the constraints in Ion Schema.
+     * @param writer is Ion Writer.
+     * @param constraints is an IonStruct which contains the top-level constraints in Ion Schema.
+     * @throws Exception if errors occur when reading or writing data.
+     */
+    public static void constructIonList(IonWriter writer, IonStruct constraints) throws Exception {
+        // When there's only one required element in Ion List and the length of generated Ion List is not specified, we set the default length as a integer smaller than 20.
+        Random random = new Random();
+        int listSize = random.nextInt(20);
+        int occurTime;
```

Comment:
"occurrences" might be a better name.

Reply:
> "occurrences" might be a better name.

Changes has been updated in the following commit.